usr/sbin/p11sak/p11sak.c

```diff
@@ -2561,13 +2589,19 @@ int main(int argc, char *argv[])
 
     /* Prompt for PIN if not already set via option */
     if (!pin) {
-        printf("Please enter user PIN:");
+        printf("Please enter user PIN: ");
         rc = get_pin(&pin, &pinlen);
+        pin_alloced = ckb_true;
```

Comment:
That seems to be wrong.  Only if `rc != -1` did we actually allocate the pin and have to free it.  But seeing that we initialize `pin` with `NULL`, the error case might not be that interesting.

Reply:
I have translate the rc to CKR_ errors anyway, so I set pin_alloced = ckb_true only for rc = 0.